persistence.js

```diff
@@ -33,6 +31,8 @@ function RedisPersistence (opts) {
   this.maxSessionDelivery = opts.maxSessionDelivery || 1000
   this._db = new Redis(opts)
 
+  this.messageIdCache = {}
```

Comment:
How can it be? 
Consider server being restarted in between a QOS>0 delivery flow
i.e. PUB -> PUBACK
or PUB -> PUBREC -> PUBREL -> PUBCOMP
why should be messageId associations preserve restart or state!? I believe messageId cache is a very volatile map lasting for a very small time period (delivering message) and if anything goes wrong, it should be discarded and delivery should be restarted on next connect!

Remember that PUBREL's are updated per client in persistence, for QOS=2

Am I missing something?